src/anomalib/data/base/video.py

```diff
@@ -68,6 +88,28 @@ def _setup_clips(self) -> None:
             frames_between_clips=self.frames_between_clips,
         )
 
+    def _select_targets(self, item):
+        if self.target_frame == VideoTargetFrame.FIRST:
+            idx = 0
+        elif self.target_frame == VideoTargetFrame.LAST:
+            idx = -1
+        elif self.target_frame == VideoTargetFrame.MID:
+            idx = int(self.clip_length_in_frames / 2)
```

Comment:
Is this how it is usually handled in the literature? Would it be an idea to just use an `index` argument to specify a specific frame rather than limiting it to first, mid and last?

Reply:
One option would be to use -1 for last frame, and -2 for all frames. Would that be an idea?

Reply:
This is probably an edge case. I'm not sure if a user would use a random index in the first place. 

If last frame is most common, then we could maybe stick to your approach. -2 for all might be confusing to users.

Reply:
I feel passing an index might be confusing. However, the only solution I can think of is to support `VideoTargetFrame.index` like this https://stackoverflow.com/questions/55996561/support-unknown-values-in-python-enums. So , we can have `VideoTargetFrame.20` to return frame number 20. But might be a bit hacky.